pxd.c

```diff
@@ -605,6 +605,7 @@ ssize_t pxd_add(struct fuse_conn *fc, struct pxd_add_out *add)
 out_id:
 	ida_simple_remove(&pxd_minor_ida, new_minor);
 out_module:
+	kfree(pxd_dev);
```

Comment:
pxd_dev could be null here (line: 558). Add this free to pxd_free_disk(), that would cover all our bases.